### PR TITLE
Roles Backend

### PR DESCRIPTION
**Pull request by sam-warren**
# Overview

This PR includes the following proposed change(s):

- Updated test users seed to add some default roles
- Added user access, user roles tables
- Add queries for user access table
- Add front end api functions to query user access 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?
Manual testing

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: database/src/migrations/20211214033115_access_request.ts [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import * as Knex from 'knex';
 
 export async function up(knex: Knex): Promise<void> {
-    await knex.raw(`
-    set schema 'invasivesbc';
+  await knex.raw(`
+    set schema 'public';
```
Comment (micheal-w-wells) [inline database/src/migrations/20211214033115_access_request.ts]: When you set a search path, you're telling psql "if I give a relation without a schema prefix, FIND it via these schemas in this order", when you give it a schema, i BELIEVE it sets the schema as where you want to CREATE a table if you don't give a schema prefix, and also doubles as a search path if you didn't give one.  Long story short I think you want to set the search path to invasivesbc or just prefix your table/relation names with invasivesbc.

File: database/src/migrations/20220106120516_user_roles.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import * as Knex from 'knex';
+const DB_SCHEMA = 'invasivesbc';
```
Comment (micheal-w-wells) [inline database/src/migrations/20220106120516_user_roles.ts:2]: this one is probably fine

File: database/src/migrations/20220113030811_access_request_patch.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import * as Knex from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`
+    set schema 'public';
```
Comment (micheal-w-wells) [inline database/src/migrations/20220113030811_access_request_patch.ts]: this one will put access_request in public

File: docker-compose.local.yml [hunk 4]
```diff
@@ -40,6 +40,7 @@ services:
       - local-network
     links:
       - db
+    restart: on-failure
```
Comment (micheal-w-wells) [inline docker-compose.local.yml:43]: nice!

**Review by micheal-w-wells — CHANGES_REQUESTED** [review]
Awesome work champ.  I think you just need to take a look at my notes for migrations and otherwise (assuming it was tested on mobile too) it should be good to go.

File: database/src/migrations/20211214033115_access_request.ts (hunk 1 above)
Comment (sam-warren) [reply database/src/migrations/20211214033115_access_request.ts]: fixed!

**Pull request merged**